### PR TITLE
[versioning][android] Support react-native 69

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

Support rn 69 in versioning script + refactor

This PR is quite large, but because update rn69 broke versioning I did not have a way to verify if my changes work without fixing everything

# How

- add debugging to transform logic, now if you add  `debug: true` to some FileTransform it will prompt every time there is a change caused by that transform and display a diff
- build versioned hermes from source
  - download hermes using existing gradle task
  - version hermes to rename the lib and to use versioned jsi
  - define env `REACT_NATIVE_OVERRIDE_HERMES_DIR` when building rn to use that directory
- version expo-modules-core (the same way as expo-av and expo-gl)
- fix expo-av and expo-gl to match versioned rn aar (I'm not sure why it worked before, the location of that aar did not change)
- add CMake versioning, the main change in the approach to versioning here is that first I'm matching specific CMake section e.g. `link_target_libraries` and later run another transform on arguments

# Test Plan

- `git clean -xf`
- `yarn && et add-sdk --platform android --sdkVersion 46.0.0`
- `./gradlew app:assembleVersionedDebug`
- install on the device
- <s>expo go launches, but is crasing on NCL when sdk 46 project is launched</s>
- NCL sdk46 does not crash natively, but it's throwing js error at the start

I'm getting the same error on unversioned project, so I'm assuming it's not versioning issue

```
TypeError: undefined is not a constructor (evaluating 'new _reactNativeReanimated.default.Value(0)')
at /home/wojtek/expo/expo/react-native-lab/react-native/Libraries/Core/ExceptionsManager.js:95:4 in reportException
at /home/wojtek/expo/expo/react-native-lab/react-native/Libraries/Core/ExceptionsManager.js:141:19 in handleException
at /home/wojtek/expo/expo/react-native-lab/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
at /home/wojtek/expo/expo/node_modules/@react-native/polyfills/error-guard.js:49:36 in ErrorUtils.reportFatalError
at /home/wojtek/expo/expo/node_modules/metro-runtime/src/polyfills/require.js:203:6 in guardedLoadModule
at http://192.168.0.10:19000/__generated__/AppEntry.bundle?platform=android&dev=true&hot=false&strict=false&minify=false:506828:3 in global code
```

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/versioning/android/utils.ts [hunk 1]
```diff
@@ -1,3 +1,11 @@
+/**
+ * Find lines matched by startRegex and endRegexp and remove all the
+ * lines between. Note it will remove entire line matched by startRegex
+ * and endRegexp regardless if pattern match entire lien or not
```
Comment (Kudo) [inline tools/src/versioning/android/utils.ts:4]:
```suggestion
 * and endRegexp regardless if pattern match entire line or not
```
typo?

**Review by Kudo — COMMENTED** [review]
first run briefly review looks good to me. tbh it's hard to review so i'll pull the branch to test tomorrow.
great work. thanks

**Comment by Kudo** [comment]
the `new _reactNativeReanimated.default.Value(0)` is coming from inconsistent reanimated/gesture-handler between expo go and home js. the issue should be addressed. i can confirm the versioned code works fine by downgrading reanimated/gesture-handler.
great work thanks @wkozyra95 for the help 😁

**Review by Kudo — APPROVED** [review]

**Review by lukmccall — APPROVED** [review]
Versioning of the `expo-modules-core` looks good 🔥🔥🔥

**Review by tsapeta — APPROVED** [review]

**Pull request merged**